### PR TITLE
Return remark in rdap response when abuse mailbox validation has failed.

### PR DESCRIPTION
**Pull request by sbusk**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java [hunk 1]
```diff
@@ -161,7 +160,11 @@ public RdapObject mapHelp(final String requestUrl) {
         return mapCommons(new RdapObject(), requestUrl);
     }
 
-    private RdapObject getRdapObject(final String requestUrl, final RpslObject rpslObject, final LocalDateTime lastChangedTimestamp, @Nullable final RpslObject abuseContact) {
+    private RdapObject getRdapObject(final String requestUrl,
+                                     final RpslObject rpslObject,
+                                     final LocalDateTime lastChangedTimestamp,
+                                     @Nullable final RpslObject abuseRoleObject,
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:166]: Mixing Nullable and Optional in the same method arguments? Can we choose one style for consistency?

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java [hunk 2]
```diff
@@ -296,7 +305,12 @@ private static Remark createRemark(final RpslObject rpslObject) {
         return new Remark(descriptions);
     }
 
-    private static boolean hasRemark(final RpslObject rpslObject) {
+    private static Remark createRemark(final CIString key, final AbuseContact abuseContact) {
+        // TODO message duplicated from QueryMessages.unvalidatedAbuseCShown
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:309]: Can the TODO be resolved?

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 3]
```diff
@@ -276,7 +276,8 @@ private Response getResponse(HttpServletRequest request, List<RpslObject> result
                         getRequestUrl(request),
                         resultObject,
                         objectDao.getLastUpdated(resultObject.getObjectId()),
-                        abuseCFinder.getAbuseContactRole(resultObject)))
+                        abuseCFinder.getAbuseContactRole(resultObject),
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java:279]: Do we need two separate calls to the abuseCFinder? can you read the abuse-mailbox from the RpslObject from the  getAbuseContactRole() ?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapObjectMapperTest.java [hunk 4]
```diff
@@ -591,14 +599,49 @@ public void help() {
...
+                    "abuse-mailbox:  abuse@test.com\n" +
+                    "admin-c:        TP1-TEST\n" +
+                    "tech-c:         TP2-TEST\n" +
+                    "phone:          +31 12345678\n" +
+                    "source:         TEST"
+                ),
+                Optional.of(abuseContact)
+        );
+
+        assertThat(result.getRemarks().get(0).getDescription().get(0), is("Abuse-mailbox validation failed. Please refer to ORG-NCC1-RIPE for further information."));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapObjectMapperTest.java:634]: Can you add a negative test in a different test case? (i.e. the messages doesn't appear)

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java (hunk 3 above)
Comment (sbusk) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java]: I've added the abuse role object to the AbuseContact so we don't need multiple lookups and we have all the information we need in one place.

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java (hunk 1 above)
Comment (sbusk) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java]: We use just the AbuseContact now.

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java (hunk 2 above)
Comment (sbusk) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:305]: Done.

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapObjectMapperTest.java (hunk 4 above)
Comment (sbusk) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapObjectMapperTest.java]: I've added a bunch of test cases (message doesn't appear, responsible org not found, IT).

**Pull request merged**